servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/HttpLifecycleObserverTest.java

```diff
@@ -355,6 +355,25 @@ public Single<StreamingHttpResponse> handle(HttpServiceContext ctx,
                 serverRequestObserver, serverResponseObserver);
     }
 
+    @ParameterizedTest(name = "{displayName} [{index}] protocol={0}")
+    @EnumSource(HttpProtocol.class)
+    void badResponsePayloadBody(HttpProtocol protocol) throws Exception {
+        serviceFilterFactory(service -> new StreamingHttpServiceFilter(service) {
+            @Override
+            public Single<StreamingHttpResponse> handle(HttpServiceContext ctx,
+                                                        StreamingHttpRequest request,
+                                                        StreamingHttpResponseFactory responseFactory) {
+                StreamingHttpResponse response = responseFactory.ok();
+                response.payloadBody(Publisher.from(CONTENT).concat(Publisher.failed(DELIBERATE_EXCEPTION)));
+                return Single.succeeded(response);
+            }
+        });
+        setUp(protocol);
+        assertThrows(ExecutionException.class,
+                () -> makeRequestAndAssertResponse(SVC_ECHO, protocol, OK, CONTENT.readableBytes()));
```

Comment:
Would be great to also enhance `AbstractHttpServiceAsyncContextTest.contextPreservedOverFilterBoundaries` to validate `AsyncContext` propagation for failure scenario

Reply:
Suggestions welcome. 😄

Reply:
To add a similar test that will run the same variations with a server endpoint that returns a failed publisher, like in your other tests, rather than a successful response.

Reply:
I added test coverage and discussion of the details is ongoing in other threads so I'm closing this one..